apps/shared/avifpng.c

```diff
@@ -312,6 +307,86 @@ avifBool avifPNGRead(const char * inputFilename,
         *outPNGDepth = imgBitDepth;
     }
 
+    if (!ignoreProfile) {
+        char * iccpProfileName = NULL;
+        int iccpCompression = 0;
+        unsigned char * iccpData = NULL;
+        png_uint_32 iccpDataLen = 0;
+        int srgbIntent = 0;
+
+        // PNG specification 4.2.2: The sRGB and iCCP chunks should not both appear.
+        // When the sRGB / iCCP chunk is present applications that recognize it and are capable of color management
+        // must ignore the gAMA and cHRM chunks and use the sRGB / iCCP chunk instead.
+        if (png_get_iCCP(png, info, &iccpProfileName, &iccpCompression, &iccpData, &iccpDataLen) == PNG_INFO_iCCP) {
+            avifImageSetProfileICC(avif, iccpData, iccpDataLen);
+        } else if (png_get_sRGB(png, info, &srgbIntent) == PNG_INFO_sRGB) {
+            // srgb_intent ignored
+            avif->colorPrimaries = AVIF_COLOR_PRIMARIES_BT709;
+            avif->transferCharacteristics = AVIF_TRANSFER_CHARACTERISTICS_SRGB;
+        } else {
+            avifBool needCraftICC = AVIF_FALSE;
+            double gamma;
+            double wX, wY, rX, rY, gX, gY, bX, bY;
+            float primaries[8];
+            if (png_get_gAMA(png, info, &gamma) == PNG_INFO_gAMA) {
+                gamma = 1.0 / gamma;
+                if (matchesTo3RoundedPlaces(gamma, 2.2)) {
+                    avif->transferCharacteristics = AVIF_TRANSFER_CHARACTERISTICS_BT470M;
+                } else if (matchesTo3RoundedPlaces(gamma, 1.0)) {
+                    avif->transferCharacteristics = AVIF_TRANSFER_CHARACTERISTICS_LINEAR;
+                } else if (matchesTo3RoundedPlaces(gamma, 2.8)) {
+                    avif->transferCharacteristics = AVIF_TRANSFER_CHARACTERISTICS_BT470BG;
+                } else {
+                    needCraftICC = AVIF_TRUE;
+                }
+            } else {
+                gamma = 2.2;
```

Comment:
Nit: Add a comment to justify this default gamma value.

Add a similar comment to line 361.

I think these default values come from http://www.libpng.org/pub/png/spec/1.2/PNG-Chunks.html#C.sRGB ?